core/src/main/java/org/neo4j/ogm/metadata/ClassInfo.java

```diff
@@ -1098,16 +1111,28 @@ public FieldInfo getEndNodeReader() {
      * @return a FieldInfo for the field annotated as the StartNode, or none if not found
      */
     public FieldInfo getStartNodeReader() {
+        if (startNodeReader == null) {
+            initStartNodeReader();
+        }
```

Comment:
```suggestion
        initStartNodeReader();
```

Reply:
Are you sure? I'm asking because I have applied the same logic than #850. I would propose to refactor the old way of doing it to remove the synchronization possible contention.

Reply:
I missed it there :) 

The additional if won't probably matter much, the JVM is going to remove it probably anyway at some point. 

I'm gonna bring this in now and check all of those in an additional change.

Reply:
Should I modify according to your [comment](https://github.com/neo4j/neo4j-ogm/pull/855#discussion_r512212383) on the #855 ?